apps/studio/src/components/sidebar/DetailViewSidebar.vue

```diff
@@ -157,18 +158,31 @@ export default Vue.extend({
     },
     processedValue() {
       const clonedValue = _.cloneDeep(this.value)
-      eachPaths(clonedValue, (path, value: string) => {
-        if (this.truncatedPaths.includes(path)) {
-          _.set(clonedValue, path, value.slice(0, globals.maxDetailViewTextLength))
+      eachPaths(clonedValue, (path, value) => {
+        if (this.jsonFields.includes(path)) {
+          _.set(clonedValue, path, JSON.parse(value as string))
```

Comment:
is value always a string? Can it ever be an object / array at this point?

Reply:
What if this fails?

Reply:
It shouldn't fail since all values in `jsonFields` are filtered with `getLanguageByContent` which uses `JSON.parse` to check whether it is a valid json string or not. But I agree that even though that's true, this still looks vulnerable.